vim-patch:8.2.{4702,4703}: C++ scope labels are hard-coded

tom-anders opened the pull request:
Problem:    C++ scope labels are hard-coded.
Solution:   Add 'cinscopedecls' to define the labels. (Tom Praschan,
            closes vim/vim#10109)
https://github.com/vim/vim/commit/3506cf34c17c5eae6c2d1317db1fcd5a8493c288

tom-anders committed:
vim-patch:8.2.4702: C++ scope labels are hard-coded

Problem:    C++ scope labels are hard-coded.
Solution:   Add 'cinscopedecls' to define the labels. (Tom Praschan,
            closes vim/vim#10109)
https://github.com/vim/vim/commit/3506cf34c17c5eae6c2d1317db1fcd5a8493c288

zeertzjq commented on runtime/doc/options.txt:
```diff
@@ -1288,7 +1288,17 @@ A jump table for the options with a short description can be found at |Q_op|.
+<						*'clipboard'* *'cb'*
 'clipboard' 'cb'	string	(default "")
 			global
 	This option is a list of comma separated names.
```
This was in the patch.
```suggestion
	This option is a list of comma-separated names.
```

zeertzjq commented on src/nvim/indent_c.c:
```diff
@@ -513,23 +513,28 @@ static int cin_isdefault(const char_u *s)
- * Recognize a "public/private/protected" scope declaration label.
+ * Recognize a scope declaration label set in 'cinscopedecls'.
  */
 bool cin_isscopedecl(const char_u *s)
```
Rename to `p`

tom-anders replied:
Sorry, rename what exactly?

zeertzjq replied:
The argument is renamed from `s` to `p` in the patch.

zeertzjq reviewed (commented):
<https://github.com/vim/vim/commit/cb49a1d934d38cbc0d445af59b71ffe45cbcae75> is also needed.

tom-anders commented:
@zeertzjq is it okay if I squash the memory leak fix into this commit? Or should I keep it separate?

zeertzjq commented:
I think it is better to port it in a separate vim-patch commit using `vim-patch.sh`.

tom-anders committed:
vim-patch:8.2.4703: memory leak in handling 'cinscopedecls'

Problem:    Memory leak in handling 'cinscopedecls'.
Solution:   Free the memory before returning.
https://github.com/vim/vim/commit/cb49a1d934d38cbc0d445af59b71ffe45cbcae75

zeertzjq commented on src/nvim/indent_c.c:
```diff
@@ -513,23 +513,31 @@ static int cin_isdefault(const char_u *s)
 
 /*
- * Recognize a "public/private/protected" scope declaration label.
+ * Recognize a scope declaration label set in 'cinscopedecls'.
```
Comment style needs to be changed to `///`-style. Since this comment change was not in the patches, wait for <https://github.com/vim/vim/pull/10126> first.

tom-anders replied:
Done
